views/application.py

```diff
@@ -93,3 +93,28 @@ def update(self, request, *args, **kwargs):
                 list(set(new_team_member_ids)-set(old_team_member_ids))
             )
         return response.Response(serializer.data)
+
+    
+class ApplicationHiddenDetailView(generics.GenericAPIView):
+    """
+        view details of application which should not be exposed directly 
+    """
+
+    permission_classes = [permissions.IsAuthenticated, ]
+
+    def post(self,request,*args,**kwargs):
+        application = Application.objects.get(pk=kwargs['id'])
+        try:
+            application = Application.objects.get(pk=kwargs['id'])
+            application_team_members = application.team_members.all()
+            for team_member in application_team_members:
+                if team_member.id == request.person.id:
+                    if request.user.check_password(request.data['password']):
+                        return response.Response(ApplicationHiddenDetailSerializer(application).data)
+                    else:
+                        return HttpResponse("Wrong Password",status=403)
```

Comment:
Use `response.Response` here as well. No need to import `HttpResponse`. Accordingly, remove the import statement as well.